pl_bolts/models/gans/srgan/components.py

```diff
@@ -0,0 +1,150 @@
+"""
+based on
+https://colab.research.google.com/github/https-deeplearning-ai/GANs-Public/blob/master/C3W2_SRGAN_(Optional).ipynb
+"""
+import torch
+import torch.nn as nn
+
+from pl_bolts.utils import _TORCHVISION_AVAILABLE
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+if _TORCHVISION_AVAILABLE:
+    from torchvision.models import vgg19
+else:
+    warn_missing_pkg("torchvision")  # pragma: no-cover
+
+
+class ResidualBlock(nn.Module):
+    def __init__(self, feature_maps: int = 64) -> None:
+        super().__init__()
+
+        # Residual block: k3n64s1 x2
+        self.block = nn.Sequential(
+            self._make_conv_block(feature_maps),
+            self._make_conv_block(feature_maps, prelu=False),
+        )
+
+    @staticmethod
+    def _make_conv_block(feature_maps: int, prelu: bool = True) -> nn.Sequential:
+        return nn.Sequential(
+            nn.Conv2d(feature_maps, feature_maps, kernel_size=3, padding=1),
+            nn.BatchNorm2d(feature_maps),
+            nn.PReLU() if prelu else nn.Identity(),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return x + self.block(x)
+
+
+class SRGANGenerator(nn.Module):
```

Comment:
I think we can do something as I mentioned in DCGAN #403 . Marking again for discussion in #427